lib/pbench/test/unit/server/query_apis/test_datasets_detail.py

```diff
@@ -19,20 +22,17 @@ class TestDatasetsDetail(Commons):
     def _setup(self, client):
         super()._setup(
             cls_obj=DatasetsDetail(client.config, client.logger),
-            pbench_endpoint="/datasets/detail/fio",
+            pbench_endpoint="/datasets/detail/random_md5_string1",
             elastic_endpoint="/_search?ignore_unavailable=true",
-            payload={
-                "user": "drb",
-                "access": "private",
-                "start": "2020-08",
-                "end": "2020-10",
-            },
+            index_from_metadata="run-data",
             empty_es_response_payload=self.EMPTY_ES_RESPONSE_PAYLOAD,
         )
 
+    api_method = API_METHOD.GET
+
     @pytest.mark.parametrize(
         "user",
-        ("drb", "badwolf", "no_user"),
+        ("drb", "no_user"),
```

Comment:
I thought we were testing a matrix, here:
| user \ access | public | private |
| --- | --- | --- |
| **owner** | ✅ | ✅ |
| **other** | ✅ | ⛔ (not allowed) |
| *none* | ✅ | 🚫 (not tested, apparently)|

You have removed the **other** line from the matrix with this change and the deletion of line 57...was that intentional?

Reply:
Good point, although the test is a bit more complicated now since we rely on actual dataset ownership rather than just a keyword Elasticsearch term. But, yes; we want to make sure that we reject a `/datasets/detail` for a dataset owned by `drb` when the authenticated user is `test`.

Reply:
> we want to make sure that we reject a /datasets/detail for a dataset owned by drb when the authenticated user is test.

@siddardh-ra, do you know how to make that happen?  🙂

Reply:
> @siddardh-ra, do you know how to make that happen? slightly_smiling_face

Yeah @webbnh. And in the recent commit, I've added test cases for all the 3 access types